### DISCUSSION TITLE
Merge seurat object

### DISCUSSION THREAD

**Discussion by dongmin7591** [satijalab/seurat#8254] [Q&A]
![스크린샷 2024-01-04 145557](https://github.com/satijalab/seurat/assets/80022367/4ffabb4c-5c33-4ca9-adcc-e64cc83d9e31)

I recently updated to seurat v5. Previously, when version 4.3 was used, the merged seurat object created after merging was divided into one layers (counts, data), but in seurat 5, counts.1,2,3,... or data1,2,3,... depending on the number of each sample. Why is expression data not merged? Is there another way?

**Comment by Dooo0k** [comment]
Use the JoinLayers() function to merge multiple layers into a single layer.
ex) obj <- JoinLayers(obj)

**Reply by alekseybelikov** [reply]
this function doesn't work. I still have separate layers after using it

**Reply by alekseybelikov** [reply]
```
merged_seurat_2 <- JoinLayers(
  object = merged_seurat,
  layers = c(
    "counts.Sample19M_RT00375N",
    "counts.Sample29M_RT00376N",
    "counts.Sample40M_RT00391N",
    "counts.Sample41M_RT00377N",
    "counts.Sample44M1_RT00383N",
    "counts.Sample44M2_RT00388N",
    "counts.Sample45M_RT00374N",
    "counts.Sample47M_RT00373N",
    "counts.Sample52M_RT00387N",
    "counts.Sample60M_RT00382N",
    "counts.Sample66M_RT00384N",
    "counts.Sample79M_RT00378N"
  )
)

Layers(merged_seurat_2)

```

'counts.Sample19M_RT00375N''counts.Sample29M_RT00376N''counts.Sample40M_RT00391N''counts.Sample41M_RT00377N''counts.Sample44M1_RT00383N''counts.Sample44M2_RT00388N''counts.Sample45M_RT00374N''counts.Sample47M_RT00373N''counts.Sample52M_RT00387N''counts.Sample60M_RT00382N''counts.Sample66M_RT00384N''counts.Sample79M_RT00378N'
